contracts/pool/FeesEscrow.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+pragma solidity 0.7.5;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
+import "../interfaces/IFeesEscrow.sol";
+import "../interfaces/IBalancer.sol";
+import "../interfaces/IWrapper.sol";
+
+/**
+ * @title FeesEscrow
+ *
+ * @dev FeesEscrow contract is used to receive tips from validators and move them to the Pool contract
+ * when calling transferToPool method.
+ */
+contract FeesEscrow is IFeesEscrow, IBalancerStruct {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    // @dev Pool contract's address.
+    address private pool;
+
+    // @dev RewardToken contract's address.
+    address private rewardToken;
+
+    // @dev Balancer pool id for Symmetric Vault.
+    bytes32 private symmetricPoolId;
+
+    // @dev Symmetric Balancer Vault contract address.
+    IBalancerVault private symmetricVault;
+
+    // @dev Native xDAI wrapper contract address.
+    address private wrapper;
+
+    // @dev GNO token contract address.
+    address private gnoToken;
+
+    constructor(
+        address _pool,
+        address _rewardToken,
+        bytes32 _symmetricPoolId,
+        IBalancerVault _symmetricVault,
+        address _wrapper,
+        address _gnoToken
+    ) {
+        pool = _pool;
+        rewardToken = _rewardToken;
+        symmetricPoolId = _symmetricPoolId;
+        symmetricVault = _symmetricVault;
+        wrapper = _wrapper;
+        gnoToken = _gnoToken;
+    }
+
+    /**
+     * @dev See {IFeesEscrow-transferToPool}.
+     */
+    function transferToPool() external override returns (uint256) {
+        require(msg.sender == rewardToken, "FeesEscrow: invalid caller");
+
+        // Wrap all accumulated native xDAI tokens to WXDAI
+        uint256 balance = address(this).balance;
+
+        if (balance == 0) {
+            return 0;
+        }
+
+        IWrapper(wrapper).deposit{value: balance}();
+
+        // Prepare data to exchange WXDAI to GNO tokens via Balancer Vault
+        bytes memory userData = new bytes(0);
+        FundManagement memory funds = FundManagement(address(this), false, address(this), false);
```

Comment:
As this variable is constant, we can make it immutable and move to the state.

Reply:
Moved, but couldn't make it immutable because only primitive types supported (and strings) to be immutable. When I try to make it immutable I receive error:
```
contracts/pool/FeesEscrow.sol:39:5: TypeError: Immutable variables cannot have a non-value type.
    IBalancerVault.FundManagement private immutable funds;
```

Also I [googled](https://ethereum.stackexchange.com/questions/127622/typeerror-immutable-variables-cannot-have-a-non-value-type) this issue and found that only primitive and string types supported for being immutable